rand-core/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+/// 
+/// Normally users should use the [`NewRng`] extension trait, excepting when a
+/// fixed seed must be used, in which case usage of [`SeedableRng::from_seed`]
+/// is recommended.
+/// 
+/// [`NewRng`]: https://docs.rs/rand/0.5/rand/trait.NewRng.html
+/// [`SeedableRng::from_seed`]: #tymethod.from_seed
+pub trait SeedableRng: Sized {
+    /// Seed type, which is restricted to types mutably-dereferencable as `u8`
+    /// arrays (we recommend `[u8; N]` for some `N`).
+    ///
+    /// It is recommended to seed PRNGs with a seed of at least circa 100 bits,
+    /// which means an array of `[u8; 12]` or greater to avoid picking RNGs with
+    /// partially overlapping periods.
+    ///
+    /// For cryptographic RNG's a seed of 256 bits is recommended, `[u8; 32]`.
+    type Seed: Sized + Default + AsMut<[u8]>;
+
+    /// Create a new PRNG using the given seed.
+    ///
+    /// PRNG implementations are allowed to assume that bits in the seed are
+    /// well distributed. That means usually that the number of one and zero
+    /// bits are about equal, and values like 0, 1 and (size - 1) are unlikely.
+    ///
+    /// PRNG implementations are recommended to be reproducible. A PRNG seeded
+    /// using this function with a fixed seed should produce the same sequence
+    /// of output in the future and on different architectures (with for example
+    /// different endianness).
+    ///
+    /// It is however not required that this function yield the same state as a
+    /// reference implementation of the PRNG given equivalent seed; if necessary
+    /// another constructor can be used.
+    fn from_seed(seed: Self::Seed) -> Self;
+
+    /// Create a new PRNG seeded from another `Rng`.
+    ///
+    /// This is the recommended way to initialize PRNGs with fresh entropy. The
+    /// [`NewRng`] trait provides a convenient new method based on `from_rng`.
+    /// 
+    /// Usage of this method is not recommended when reproducibility is required
+    /// since implementing PRNGs are not required to fix Endianness and are
+    /// allowed to modify implementations in new releases.
+    ///
+    /// It is important to use a good source of randomness to initialize the
+    /// PRNG. Cryptographic PRNG may be rendered insecure when seeded from a
+    /// non-cryptographic PRNG or with insufficient entropy.
+    /// Many non-cryptographic PRNGs will show statistical bias in their first
+    /// results if their seed numbers are small or if there is a simple pattern
+    /// between them.
+    ///
+    /// Prefer to seed from a strong external entropy source like [`OsRng`] or
+    /// from a cryptographic PRNG; if creating a new generator for cryptographic
+    /// uses you *must* seed from a strong source.
+    ///
+    /// Seeding a small PRNG from another small PRNG is possible, but
+    /// something to be careful with. An extreme example of how this can go
+    /// wrong is seeding an [`XorShiftRng`] from another [`XorShiftRng`], which
```

Comment:
Do we still want this link to `XorShiftRng`? I removed it in the original PR because of the crate separation.

Reply:
I decided to keep this paragraph as is since it provides a clear warning and only mentions Xorshift as an example (we could even remove the link and it would still be clear).

This does leave a couple of cross-crate doc links, but I don't see a major problem (as long as we keep the number small).